lib/user.rb

```diff
@@ -15,14 +15,17 @@ def defaults
   end
 
   def addsong spotify_id
-    song = Spotify.find_song spotify_id
-    artist = song[:artist]
-    track = song[:track_name]
-    preview_link = song[:preview_link]
-    play_link = song[:play_link]
-
-    # User spends vote here
-    Song.create!(title: track, artist: artist, spotify_id: spotify_id, user_id: self.id, preview_link: preview_link, play_link: play_link)
+    song = Spotify.find_song spotify_id 
```

Comment:
This works, but I think would be slightly cleaner if `find_song` returned `nil` on error. You could skip the `else` branch in that function, and write this as:
```
if song = Spotify.find_song spotify_id
   ...
end
```